packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverButton/TeachingPopoverButton.types.ts

```diff
@@ -1,17 +1,30 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import type { ButtonProps, ButtonSlots, ButtonState } from '@fluentui/react-button';
 
-export type TeachingPopoverButtonSlots = {
-  root: Slot<'div'>;
-};
-
-/**
- * TeachingPopoverButton Props
+/*
+ * These buttons are the same as a default fluent v9 button
+ * with overridden style/functionality based on Teaching Context
  */
-export type TeachingPopoverButtonProps = ComponentProps<TeachingPopoverButtonSlots> & {};
+export type TeachingPopoverButtonSlots = ButtonSlots;
 
-/**
- * State used in rendering TeachingPopoverButton
- */
-export type TeachingPopoverButtonState = ComponentState<TeachingPopoverButtonSlots>;
-// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from TeachingPopoverButtonProps.
-// & Required<Pick<TeachingPopoverButtonProps, 'propName'>>
+export type TeachingPopoverButtonType = 'primary' | 'secondary';
+
+export type TeachingPopoverButtonProps = ButtonProps & {
+  /*
+   * Button type, compulsory to be provided by user.
+   */
+  buttonType: TeachingPopoverButtonType;
+
+  /*
+   * Optional text shown when a primary button is on it's final step
+   * OR when a secondary button is on it's first step
+   */
+  altStepText?: string;
```

Comment:
I don't think that it's the most flexible approach, let's add an item to investigate other options

Reply:
Resolution tracked by #30127.